src/PSRule.Rules.Azure/rules/Azure.Deployment.Rule.ps1

```diff
@@ -55,4 +60,52 @@ function global:RecurseDeploymentSensitive {
     }
 }
 
+# Check resource properties that should be set by secure parameters.
+function global:RecurseSecureValue {
+    param (
+        [Parameter(Mandatory = $True)]
+        [PSObject]$Deployment
+    )
+    process {
+        $resources = @($Deployment.properties.template.resources);
+        if ($resources.Length -eq 0) {
+            return $Assert.Pass();
+        }
+
+        $secureParameters = @($Deployment.properties.template.parameters.PSObject.properties | Where-Object {
+            $_.Value.type -eq 'secureString' -or $_.Value.type -eq 'secureObject'
+        } | ForEach-Object {
+            $_.Name
+        });
+        Write-Debug -Message "Secure parameters are: $($secureParameters -join ', ')";
+
+        foreach ($resource in $resources) {
+            if ($resource.type -eq 'Microsoft.Resources/deployments') {
+                RecurseSecureValue -Deployment $resource;
+            }
+            elseif ($resource.type -eq 'Microsoft.Compute/virtualMachineScaleSets') {
+                if ($resource.properties.virtualMachineProfile.osProfile.adminPassword) {
+                    $isSecure = [PSRule.Rules.Azure.Runtime.Helper]::HasValueFromSecureParameter($resource.properties.virtualMachineProfile.osProfile.adminPassword, $secureParameters);
```

Comment:
We should probably avoid flagging the property when it was set using keyVault or listKeys, right? Looking for that placeholder value you told me about

Reply:
@verabe You are correct. Good catch. Do you want to have a go updating the validator?

Reply:
Sure